src/transaction/txParser.ts

```diff
@@ -183,26 +237,47 @@ const parseTxCerts = (txCertificates: any[]): _Certificate[] => {
   )
 }
 
-const parseTxWithdrawals = (
-  withdrawals: any,
-): _Withdrawal[] => {
+const parseTxWithdrawals = (withdrawals: any): _Withdrawal[] => {
   if (!isWithdrawalsMap(withdrawals)) {
     throw Error(Errors.WithrawalsParseError)
   }
   return Array.from(withdrawals).map(([address, coins]): _Withdrawal => ({ address, coins: BigInt(coins) }))
 }
 
-const parseTtl = (ttl: any): number => {
+const parseFee = (fee: any): BigInt => {
+  if (!isBigNumber(fee)) {
+    throw Error(Errors.FeeParseError)
+  }
+  return BigInt(fee)
+}
+
+const parseTtl = (ttl: any): number | undefined => {
   if (ttl === null || ttl === undefined) {
-    throw Error(Errors.MssingTTLParseError)
+    return undefined
+  }
+  if (!Number.isInteger(ttl)) {
+    throw Error(Errors.TTLParseError)
   }
   return ttl
 }
 
+const parseValidityIntervalStart = (validityIntervalStart: any): BigInt | undefined => {
```

Comment:
if the the current slot/time ratio doesnt change very much or cardano-hw-cli isnt here in milion-trilion years, validity interval and ttl should just numbers not BigInts.

Reply:
Fixed